### PR TITLE
Added result card Filters

### PR DESCRIPTION
**Pull request by raoaliahmad**

### CODE REVIEW

File: web/src/modules/Student/Single/Marks/index.js [hunk 1]
```diff
@@ -187,11 +187,11 @@ export const StudentMarks = ({student, exams, settings, startDate=0, endDate=mom
 		</div>
 	
 		<div className="print-only">
-			<div style={{ marginTop: "100px" }}>
+			<div style={{ marginTop: "80px" }}>
 				<div>Signature: ___________________</div>
 			</div>
 
-			<div style={{ marginTop: "50px" }}>
+			<div style={{ marginTop: "40px", marginBottom:"40px" }}>
```
Comment (raoaliahmad) [inline web/src/modules/Student/Single/Marks/index.js]: @taimur38 this works fine I doesn't takes the report to the next page. You should also test it.

File: web/src/modules/Student/Single/Marks/index.js [hunk 2]
```diff
@@ -58,7 +93,22 @@ class StudentMarksContainer extends Component {
...
+	}
+	else if(subjectFilter !== ""){
+		return subjectFilter === subject
+	}
+	else
+		return true
+	
+}
+
+export const reportStringForStudent = (student, exams, startDate=0, endDate=moment.now(), examFilter, subjectFilter) => {
```
Comment (taimur38) [inline web/src/modules/Student/Single/Marks/index.js]: default function values should only be at the end of the function signature

File: web/src/modules/Student/Single/Marks/index.js [hunk 3]
```diff
@@ -58,7 +93,22 @@ class StudentMarksContainer extends Component {
...
+		return examFilter === exam && subjectFilter === subject 
+	else if(examFilter !== ""){
+		return examFilter === exam
+	}
+	else if(subjectFilter !== ""){
+		return subjectFilter === subject
+	}
+	else
+		return true
+
```
Comment (taimur38) [inline web/src/modules/Student/Single/Marks/index.js]: ` (examFilter === exam || examFilter === "") && (subjectFilter === subject || subjectFilter === "")`

File: web/src/modules/Student/Single/Marks/index.js (hunk 1 above)
Comment (taimur38) [reply web/src/modules/Student/Single/Marks/index.js]: fine for now

File: web/src/modules/Student/Single/Marks/index.js [hunk 4]
```diff
@@ -93,22 +93,15 @@ class StudentMarksContainer extends Component {
...
-	else if(subjectFilter !== ""){
-		return subjectFilter === subject
-	}
-	else
+export const getReportFilterCondition = (examFilter, exam, subjectFilter, subject) => 
+{
+	if((examFilter === exam || examFilter === "") && (subjectFilter === subject || subjectFilter === "")){
 		return true
-	
+	}
```
Comment (taimur38) [inline web/src/modules/Student/Single/Marks/index.js]:
```javascript
return (examFilter === exam || examFilter === "") && (subjectFilter === subject || subjectFilter === "") ;
```

File: web/src/modules/Student/Single/Marks/index.js [hunk 5]
```diff
@@ -93,22 +93,15 @@ class StudentMarksContainer extends Component {
...
+
 }
 
-export const reportStringForStudent = (student, exams, startDate=0, endDate=moment.now(), examFilter, subjectFilter) => {
```
Comment (taimur38) [inline web/src/modules/Student/Single/Marks/index.js]: keep this order. just set default value = "" for examFilter and subjectFilter

**Pull request merged**